letsencrypt/client/standalone_authenticator.py

```diff
@@ -266,30 +266,23 @@ def already_listening(self, port):  # pylint: disable=no-self-use
         If so, also tell the user via a display notification.
 
         .. warning::
-            The current implementation is Linux-specific.  (On other
-            operating systems, it will simply not detect bound ports.)
-            This function can only usefully be run as root.
+            On some operating systems, this function can only usefully be
+            run as root.
 
         :param int port: The TCP port in question.
         :returns: True or False."""
 
+        listeners = [conn.pid for conn in psutil.net_connections() \
+                     if conn.status == 'LISTEN' and \
+                     conn.type == socket.SOCK_STREAM and \
+                     (conn.laddr == ('0.0.0.0', port) or \
+                     conn.laddr == ('::', port))]
         try:
-            proc = subprocess.Popen(
-                [constants.NETSTAT, "-nta", "--program"],
-                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            stdout, _ = proc.communicate()
-            if proc.wait() != 0:
-                raise OSError("netstat subprocess failed")
-            lines = [x.split() for x in stdout.split("\n")[2:] if x]
-            listeners = [L[6] for L in lines if
-                         # IPv4 socket case
-                         (L[0] == 'tcp' and L[5] == 'LISTEN' \
-                         and L[3] == '0.0.0.0:{0}'.format(port)) or \
-                         # IPv6 socket case
-                         (L[0] == 'tcp6' and L[5] == 'LISTEN' \
-                         and L[3] == ':::{0}'.format(port))]
-            if listeners:
-                pid, name = listeners[0].split("/")
+            if listeners and listeners[0]:
```

Comment:
Please test for `None` explicitly: `if listeners and listeners[0] is not None`.

Anyway, it's unclear to me why do you build a **list** of `listeners` if you use only the first one in the end.  If there is no reason, then code should be simplified. Otherwise explanatory docs should be added. Is it possible that `len(listeners) > 1`?